src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/cost/WheelchairCostCalculator.java

```diff
@@ -0,0 +1,93 @@
+package org.opentripplanner.routing.algorithm.raptoradapter.transit.cost;
+
+import static org.opentripplanner.model.WheelChairBoarding.NOT_POSSIBLE;
+import static org.opentripplanner.model.WheelChairBoarding.NO_INFORMATION;
+import static org.opentripplanner.model.WheelChairBoarding.POSSIBLE;
+
+import javax.annotation.Nonnull;
+import org.opentripplanner.routing.api.request.WheelchairAccessibilityRequest;
+import org.opentripplanner.transit.raptor.api.transit.CostCalculator;
+import org.opentripplanner.transit.raptor.api.transit.RaptorTransfer;
+import org.opentripplanner.transit.raptor.api.transit.RaptorTransferConstraint;
+import org.opentripplanner.transit.raptor.api.transit.RaptorTripSchedule;
+
+public class WheelchairCostCalculator implements CostCalculator {
+
+  private final CostCalculator delegate;
+  private final int[] wheelchairBoardingCost;
+
+  public WheelchairCostCalculator(
+    @Nonnull CostCalculator delegate,
+    @Nonnull WheelchairAccessibilityRequest requirements
+  ) {
+    // assign the costs for boarding a trip with the following accessibility values
+    wheelchairBoardingCost = new int[3];
```

Comment:
Could we use an `EnumMap` here for type safety?

Reply:
@t2gran Has requested this to be an array for very fast access.

Reply:
Ok, EnumMap is basically an array with some null checks. Of course it only stores the boxed type, so it will not be exactly the same performance, but I'd expect the impact to be low. Maybe we could have an EnumIntMap in OTP in the future for these kind of cases.